data/transactions/logic/eval.go

```diff
@@ -4062,6 +4062,13 @@ func opTxBegin(cx *EvalContext) {
 		cx.err = errors.New("itxn_begin without itxn_submit")
 		return
 	}
+
+	// Product decision - Prevent clear state programs (CSPs) from issuing inner
+	// transactions to minimize complexity.
+	// * If CSPs support inner transactions, downstream program errors may
+	//   complicate CSP guarantees for opting out accounts from apps.
```

Comment:
I don't think that's real reason.  It has more to do with the inability to properly account for fees paid for transactions that never end up committing (because the CSP failed, so its effects must rollback).  Some of the possible solutions to _that_ problem have have complications that would make it difficult for accounts to opt out of an account.

All in all, I think explaining the rationale would take a page of code and maybe isn't justified.

Reply:
Had an out-of-band discussion and opting to defer on the PR for now.  Resolving.